### PR TITLE
[#261] Implemented precedent similarity using mahalanobi distance

### PR DESCRIPTION
**Pull request by arekmano**
- Returns the 5 most similar cases by the case's features (demands, facts, outcomes)
- The better our feature extraction is, the better the similarity metric will be.

error. commit message says that it uses KNN, yet this is false. Algorithm computes distance metric for all points and returns the ones with smallest distance. (see http://scikit-learn.org/stable/modules/generated/sklearn.neighbors.NearestNeighbors.html#sklearn.neighbors.NearestNeighbors.kneighbors)

[#261]

### CODE REVIEW

**Review by Samuel-Campbell — APPROVED** [review]

**Review by Samuel-Campbell — COMMENTED** [review]
Good idea to pre train the similar cases

**Comment by arekmano** [comment]
In order to evaluate the best distance metric, manual precedent evaluation must be performed.

After this, we can determine how accurate the measurement is in detecting similarity
